addons/mail/static/src/owl/components/notification_list/notification_list.js

```diff
@@ -52,12 +52,38 @@ class NotificationList extends Component {
      */
     _useStore(state, props) {
         const threads = this._useStoreThreads(state, props);
-        const notifications = threads.map(thread => {
-            return {
-                threadLocalId: thread.localId,
-                type: 'thread',
-                uniqueId: thread.localId,
-            };
+        const notifications = threads
+            .sort((threadA, threadB) => {
+                if (
+                    threadA.message_unread_counter > 0
+                    && threadB.message_unread_counter === 0
+                ) {
+                    return -1;
+                } else if (
+                    threadA.message_unread_counter === 0
+                    && threadB.message_unread_counter > 0
+                ) {
+                    return 1;
+                }
+                const messageLocalIdsA = threadA.messageLocalIds;
+                const messageLocalIdsB = threadB.messageLocalIds;
+                const lastMessageA = state.messages[messageLocalIdsA[messageLocalIdsA.length-1]];
+                const lastMessageB = state.messages[messageLocalIdsB[messageLocalIdsB.length-1]];
+                if (lastMessageA && lastMessageB) {
+                    return lastMessageB.date - lastMessageA.date;
```

Comment:
It looks like moment subtraction returns difference of unix time units of the moment objects, but I fail to see this in the documentation.

I prefer something safer like `lastMessageA.date.isBefore(lastMessageB.date) ? -1 : 1`